src/storage/kv/cursor_builder.rs

```diff
@@ -72,11 +73,19 @@ impl<'a, S: 'a + Snapshot> CursorBuilder<'a, S> {
 
     /// Build `Cursor` from the current configuration.
     pub fn build(self) -> Result<Cursor<S::Iter>> {
-        let mut iter_opt = IterOption::new(
-            self.lower_bound.map(|k| k.into_encoded()),
-            self.upper_bound.map(|k| k.into_encoded()),
-            self.fill_cache,
-        );
+        let l_bound = if let Some(ref b) = self.lower_bound {
+            let builder = KeyBuilder::from_slice(b.as_encoded(), DATA_KEY_PREFIX_LEN, 0);
```

Comment:
why not use `KeyBuilder::from_vec(b.into_encoded())`?

Reply:
`KeyBuilder::from_vec(b.into_encoded())` is more slow.

Reply:
why？so you mean `ptr::copy()` is slower than `extend_from_slice()`?

Reply:
You are write, here we can move the vec, has fixed.